nomad/node_endpoint.go

```diff
@@ -135,6 +136,17 @@ func (n *Node) Register(args *structs.NodeRegisterRequest, reply *structs.NodeUp
 	if args.Node.SecretID == "" {
 		return fmt.Errorf("missing node secret ID for client registration")
 	}
+	// Validate node pool value if provided. The node is canonicalized in the
+	// FSM, where an empty node pool is set to "default".
+	if args.Node.NodePool != "" {
+		invalidNames := set.From([]string{
+			structs.NodePoolAll,
+		})
+		err := structs.ValidateNodePoolName(args.Node.NodePool, invalidNames)
```

Comment:
Are we ever going to call this without a single item set of `structs.NodePoolAll`?

Reply:
It would probably make sense to extract `invalidNames` into a "constant" `var` in `structs.go` and hide its details in `ValidateNodePoolName` - since we never call this helper with any other variation. That way we don't duplicate the creation of `invalidNames` (here and in agent/command.go)

Reply:
I thought I had a case where `default` was also not valid, but I think it may have gone away as the code evolve. I will rethink this bit 😄

Reply:
Yeah, I couldn't find the case I was thinking about. I removed the second argument and checked for `all` in the command and register endpoint directly.